Content.Client/CrewMedal/UI/CrewMedalBoundUserInterface.cs

```diff
@@ -0,0 +1,49 @@
+using Content.Shared.CrewMedal;
+using Robust.Client.UserInterface;
+
+namespace Content.Client.CrewMedal.UI;
+
+/// <summary>
+/// Initializes a <see cref="CrewMedalWindow"/> and updates it when new server messages are received.
+/// </summary>
+public sealed class CrewMedalBoundUserInterface : BoundUserInterface
+{
+    [Dependency] private readonly IEntityManager _entManager = default!;
+
+    [ViewVariables]
+    private CrewMedalWindow? _window;
+
+    public CrewMedalBoundUserInterface(EntityUid owner, Enum uiKey) : base(owner, uiKey)
+    {
+        IoCManager.InjectDependencies(this);
+    }
+
+    protected override void Open()
+    {
+        base.Open();
+
+        _window = this.CreateWindow<CrewMedalWindow>();
+
+        _window.OnReasonChanged += OnReasonChanged;
+        Reload();
+    }
+
+    private void OnReasonChanged(string newReason)
+    {
+        if (_entManager.TryGetComponent<CrewMedalComponent>(Owner, out var component) &&
+            component.Reason.Equals(newReason))
+            return;
+
+        SendPredictedMessage(new CrewMedalReasonChangedMessage(newReason));
+    }
+
+    public void Reload()
+    {
+        if (_window == null || !_entManager.TryGetComponent<CrewMedalComponent>(Owner, out var component))
+            return;
+
+        _window.SetCurrentReason(component.Reason);
+        _window.SetAwarded(component.Awarded);
```

Comment:
Should SetAwarded() be called before SetCurrentReason()? Don't know how a medal could be awarded while a player has the UI open, but if you do it in the order SetAwarded->SetCurrentReason, the award will leave focus of the LineEdit, so the SetCurrentReason will update to use the new text (which the user can't edit anyway, since it's awarded).